lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'active_support'
+require 'active_support/core_ext/object/deep_dup'
+
+# AutoHCK module
+module AutoHCK
+  # UtilsParser class
+  class UtilsParser
+    # Converts string and float input to int. Strips all non-numeric
+    # characters from strings.
+    #
+    # @param value [String, Float] Input value
+    #
+    # @return [Integer, nil] Integer if successful conversion, otherwise nil
+    def self.convert_to_int(value)
+      if value.is_a?(String)
+        str_val = value.gsub(/[^0-9\-.]/, '')
+        begin
+          Integer(str_val)
+        rescue ArgumentError, TypeError
+          begin
+            Integer(Float(str_val))
+          rescue ArgumentError, TypeError
+            nil
+          end
+        end
+      elsif value.is_a?(Numeric)
+        value.to_i
+      end
+    end
+  end
+
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      rows = CSV.parse(data, headers: true)
+      raw_data = rows.first.to_h
+
+      raw_output = {}
+      raw_data.each do |k, v|
+        raw_output[k] = case k
+                        when 'Hotfix(s)', 'Processor(s)'
+                          parse_hotfixs_or_processors(v)
```

Comment:
It seems "Network Card(s)" also use the same syntax so perhaps it's better to include it and name the method: `parse_array`.

Reply:
Similar but not the same.
Processors: `4 Processor(s) Installed.,[01]: Intel64 Family 6 Model 42 Stepping 7 GenuineIntel ~3392 Mhz,[02]: ....`
Network: `2 NIC(s) Installed.,[01]: vmxnet3 Ethernet Adapter,      Connection Name: Local Area Connection 3,      DHCP Enabled:    No,      IP address(es),      [01]: 10.20.1.10,   ...`

Network Card(s) is a complete structure with a lot of `,` separators for internal object parts.